src/FilBeamOperator.sol

```diff
@@ -0,0 +1,235 @@
+    /// @param dataSetId The data set ID to settle
+    function _settleCDNPaymentRail(uint256 dataSetId) internal {
+        DataSetUsage storage usage = dataSetUsage[dataSetId];
+
+        // Early return if data set not initialized or no usage to settle
+        if (usage.maxReportedEpoch == 0 || usage.cdnAmount == 0) {
+            return;
+        }
+
+        // Get rail ID from FWSS
+        IFWSS fwss = IFWSS(fwssContractAddress);
+        IFWSS.DataSetInfo memory dsInfo = fwss.getDataSetInfo(dataSetId);
+
+        // Early return if no CDN rail configured
+        if (dsInfo.cdnRailId == 0) {
+            return;
+        }
+
+        // Get the actual amount we can settle based on rail lockup
+        uint256 amountToSettle = _getSettleableAmount(dsInfo.cdnRailId, usage.cdnAmount);
+
+        // Early return if nothing can be settled (no lockup available)
+        if (amountToSettle == 0) {
+            return;
+        }
+
+        // Settle the amount and update remaining balance
+        fwss.settleFilBeamPaymentRails(dataSetId, amountToSettle, 0);
+        usage.cdnAmount -= amountToSettle;
+
+        emit CDNSettlement(dataSetId, amountToSettle);
+    }
+
+    /// @notice Settles cache miss payment rails for multiple data sets
+    /// @dev Anyone can call this function to trigger settlement
+    /// @param dataSetIds Array of data set IDs to settle
+    function settleCacheMissPaymentRails(uint256[] calldata dataSetIds) external {
+        for (uint256 i = 0; i < dataSetIds.length; i++) {
+            _settleCacheMissPaymentRail(dataSetIds[i]);
+        }
+    }
+
+    /// @dev Internal function to settle cache miss payment rail for a single data set
+    /// @param dataSetId The data set ID to settle
+    function _settleCacheMissPaymentRail(uint256 dataSetId) internal {
+        DataSetUsage storage usage = dataSetUsage[dataSetId];
+
+        // Early return if data set not initialized or no usage to settle
+        if (usage.maxReportedEpoch == 0 || usage.cacheMissAmount == 0) {
+            return;
+        }
+
+        // Get rail ID from FWSS
+        IFWSS fwss = IFWSS(fwssContractAddress);
+        IFWSS.DataSetInfo memory dsInfo = fwss.getDataSetInfo(dataSetId);
+
+        // Early return if no cache miss rail configured
+        if (dsInfo.cacheMissRailId == 0) {
+            return;
+        }
+
+        // Get the actual amount we can settle based on rail lockup
+        uint256 amountToSettle = _getSettleableAmount(dsInfo.cacheMissRailId, usage.cacheMissAmount);
+
+        // Early return if nothing can be settled (no lockup available)
+        if (amountToSettle == 0) {
+            return;
+        }
+
+        // Settle the amount and update remaining balance
+        fwss.settleFilBeamPaymentRails(dataSetId, 0, amountToSettle);
+        usage.cacheMissAmount -= amountToSettle;
+
+        emit CacheMissSettlement(dataSetId, amountToSettle);
+    }
```

Comment:
these two methods `_settleCDNPaymentRail()` and `_settleCacheMissPaymentRail()` are structurally the same, wdyt about refactoring into one

```
_settlePaymentRail(uint256 dataSetId, uint256 amount, uint railId)
```

and calling it like this

```
_settlePaymentRail(dataSetId, usage.cacheMissAmount, dsInfo.cacheMissRailId)
//
_settlePaymentRail(dataSetId, usage.cdnAmount, dsInfo.cdnRailId)
```

While the `usage` and `dsInfo` lookups will have to be repeated, those are trivial and more fine to repeat than the rest of the logic in this function.